samples/snippets/quickstart_json_credentials.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python
+
+# Copyright 2021 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Google Analytics Data API sample quickstart application.
+
+This application demonstrates the usage of the Analytics Data API using
+service account credentials from a JSON file downloaded from
+the Google Cloud Console.
+
+Before you start the application, please review the comments starting with
+"TODO(developer)" and update the code to use correct values.
+
+Usage:
+  pip3 install --upgrade google-analytics-data
+  python3 quickstart_json_credentials.py
+"""
+# [START google_analytics_data_quickstart]
+from google.analytics.data_v1beta import BetaAnalyticsDataClient
+from google.analytics.data_v1beta.types import DateRange
+from google.analytics.data_v1beta.types import Dimension
+from google.analytics.data_v1beta.types import Metric
+from google.analytics.data_v1beta.types import RunReportRequest
+
+def get_client(credentials_json_path=''):
+  """Creates an API client instance."""
+  # [START google_analytics_data_initialize]
+  # TODO(developer): Uncomment this variable and replace with a valid path to
+  #  the credentials.json file for your service account downloaded from the
+  #  Cloud Console.
+  #credentials_json_path = '/path/to/credentials.json'
+
+  # Explicitly use service account credentials by specifying
+  # the private key file.
+  return BetaAnalyticsDataClient().from_service_account_json(
```

Comment:
This doesn't seem need to need it's own method.

Reply:
It is kind of problematic to write a test otherwise. While I cannot test the call with an actual service account credentials JSON file, it is still possible to create a Client object using default credentials and pass it to sample_run_report() method. This way the test at least can ensure that the actual API call method is working fine. Alternatively, do you think there is a way to have a real service account JSON file available during a test somehow?

Reply:
@busunkim96 helped me to figure out how to use a service account json file which is actually provided by the CI. I moved  BetaAnalyticsDataClient().from_service_account_json to the same method with the rest of the code and updated the test to use the json credentials file available in test environment.